src/Raven.Server/Documents/Sharding/Subscriptions/SubscriptionConnectionsStateOrchestrator.cs

```diff
@@ -34,7 +34,7 @@ namespace Raven.Server.Documents.Sharding.Subscriptions;
 public class SubscriptionConnectionsStateOrchestrator : SubscriptionConnectionsStateBase<OrchestratedSubscriptionConnection>
 {
     private readonly ShardedDatabaseContext _databaseContext;
-    private Dictionary<string, ShardedSubscriptionWorker> _shardWorkers;
+    public Dictionary<string, ShardedSubscriptionWorker> ShardWorkers;
```

Comment:
I would prefer not to do that, but instead create a method here to check the status of `_shardWorkers.All(x => x.Value.ClosedDueNoDocsLeft)`

Reply:
done